generators/app/templates/src/containers/TicTacToe/GameListPage/children/List/list.component.js

```diff
@@ -214,31 +214,44 @@ const List = ({ webId, gamePath, sendNotification }: Props) => {
           gameUrls.push(value);
         }
 
+        if (gameUrls.length === 0) {
+          return gameUrls;
+        }
+
         // Run the list of games fetched from the document through the shex validator helper function and store the valid items in gameList
         const gameShapeUrl = 'https://solidsdk.inrupt.net/sdk/tictactoe.shex';
         gameList = await shexUtil.validateList(gameUrls, gameShapeUrl);
 
         let games = [];
-        for await (const item of gameList) {
-          // the url at this point the document is being cached by LDFlex, we need to manually clear it
-          ldflex.clearCache(item);
-          const game = await ldflexHelper.fetchLdflexDocument(item);
-          let gameData = { url: item };
-          if (game) {
-            for await (const field of tictactoeShape.shape) {
-              let values = [];
-              for await (const val of game[getPredicate(field)]) {
-                values = [...values, val.value];
-              }
-              const value = values.length > 1 ? values : values[0];
-              gameData = { ...gameData, [field.predicate]: value };
-            }
-            const opponent = await getPlayerInfo(gameData.opponent);
-            const actor = await getPlayerInfo(gameData.actor);
-            gameData = { ...gameData, opponent, actor, deleted: false, documentUrl: url };
-          } else {
-            gameData = { ...gameData, status: GameStatus.DELETED, deleted: true, documentUrl: url };
+
+        // Looping over all valid game data
+        for await (const game of gameList) {
+          let gameData = {};
+
+          // Looping over each predicate in the game shape
+          for (const field of tictactoeShape.shape) {
```

Comment:
It seems this whole loop is doing very specific work for something that seems like it should be very generic - i.e. taking a blob of RDF that has been validated to conform to a specific Shape, and copy all the data mentioned in the Shape from the RDF blob and copy it into a plain JavaScript Object so that it can be rendered. Could this not be factored out into a single utility function...?

Reply:
I'll add this to the tech debt list